src/lib.rs

```diff
@@ -42,123 +38,90 @@
             };
-            for dir_entry in dir_iter {
-                let entry = match dir_entry {
-                    Ok(f) => f,
-                    _ => continue,
-                };
-                let fpath = entry.path();
-                let fname = match entry.file_name().into_string() {
-                    Ok(n) => n,
-                    _ => continue,
-                };
-
-                // If hidden files not allowed, ignore dotfiles.
-                if self.ignore_dotfiles && fname.starts_with('.') {
-                    continue;
-                };
-
-                // If extensions are specified, proceed only if filename has one of the allowed
-                // extensions.
-                if !self.allowed_extensions.is_empty() {
-                    if let Some(extension) = fpath.extension() {
-                        if let Ok(extension) = &extension.to_owned().into_string() {
-                            if !self.allowed_extensions.contains(&extension) {
-                                continue;
-                            }
-                        } else {
-                            continue;
-                        }
-                    } else {
+
+            // If hidden files not allowed, ignore dotfiles.
+            if ignore_dotfiles && fname.starts_with('.') {
+                continue;
+            };
+
+            // If extensions are specified, proceed only if filename has one of the allowed
+            // extensions.
+            if !allowed_extensions.is_empty() {
+                if let Some(extension) = fpath.extension().and_then(|ext| ext.to_str()) {
+                    if !allowed_extensions.iter().map(|ae| ae.as_ref()).any(|ae| ae == extension) {
                         continue;
                     }
+                } else {
+                    continue;
                 }
+            }
 
-                // Check filetype, ignore non-file.
-                let meta = match entry.metadata() {
-                    Ok(m) => m,
-                    _ => continue,
-                };
-                if !meta.file_type().is_file() {
-                    if let Ok(target) = fs::read_link(&fpath) {
-                        // A devnull symlink is a special case to ignore previous file-names.
-                        if target == path::PathBuf::from("/dev/null") {
-                            trace!("Nulled config file '{}'", fpath.display());
-                            files_map.remove(&fname);
-                        }
+            // Check filetype, ignore non-file.
+            let meta = match entry.metadata() {
+                Ok(m) => m,
+                _ => continue,
+            };
+            if !meta.file_type().is_file() {
+                if let Ok(target) = fs::read_link(&fpath) {
+                    // A devnull symlink is a special case to ignore previous file-names.
+                    if target == path::PathBuf::from("/dev/null") {
```

Comment:
Let's not allocate a copy of `/dev/null` in a loop?  This could just be `if target.as_path() == "/dev/null".into()` no?